SA3/qa_web/views.py

```diff
@@ -79,12 +79,25 @@ def questions(request):
 
 def answers(request, id_):
     q = get_object_or_404(Questions, pk=id_)
-    if request.method == 'POST':
+    if request.method == 'POST' and 'answer_form' in request.POST: #Update's database when somebody answers a question
         form = AnswersForm(request.POST)
-        if form.is_valid():
+        if form.is_valid(): 
             Answers.objects.create(content=request.POST['content'], owner=request.user, question=q)
-    q_answers = Answers.objects.filter(question=q)
-    return render(request, 'qa_web/answerspage.html', {'currentQuestion': q, 'answers': q_answers})
+    elif request.method == 'POST' and 'deselect' in request.POST:  #Update's database when somebody deselects best answer.
+        updateAnswer = Answers.objects.get(correct_answer=True)
```

Comment:
Here I think you just forgot to explicitly make sure the corresponding question was selected in your query by doing `.objects.filter(question=q, correct_answer=True)`, like you did in line `91`, which is perfect. I'll add this fix to the next commit i will do in this PR :), so it's all good.

Reply:
I believe this was the only error, great job!

Reply:
Actually, since filter is a` QuerySet`, you also need to specify a specific object of that `QuerySe`t in order to` save() `it. Once again, I will add this fix to my commit.